src/Data/Bin/Ordering.agda

```diff
@@ -0,0 +1,790 @@
+... | tri≈ _  eq _  =  yes (≤-reflexive eq)
+... | tri> _  _  gt =  no (<⇒≱ {y} {x} gt)
+
+≤-isPreorder :  IsPreorder _≡_ _≤_
+≤-isPreorder =  record
+  { isEquivalence = isEquivalence
+  ; reflexive     = ≤-reflexive
+  ; trans         = ≤-trans
+  }
+
+<-respˡ-≈ :  _<_ Respectsˡ _≡_
+<-respˡ-≈ {x} {_} =  subst (_< x)
+
+<-respʳ-≈ :  _<_ Respectsʳ _≡_
+<-respʳ-≈ {x} {_} =  subst (x <_)
+
+<-resp-≈ :  _<_ Respects₂ _≡_
+<-resp-≈ =  ( <-respʳ-≈ , <-respˡ-≈ )
+
+------------------------------------------------------------------------------
+-- Several monotonicity proofs for _+_, suc
+------------------------------------------------------------------------------
+
+module IneqReasoning = InequalityReasoning {A = Bin} {_≈_ = _≡_} ≤-isPreorder <-trans
+                                           <-resp-≈ <⇒≤ <-≤-trans ≤-<-trans
+
++-mono-≤ :  _+_ Preserves₂ _≤_ ⟶ _≤_ ⟶ _≤_
++-mono-≤ {x} {x'} {y} {y'} x≤x' y≤y' =  begin
+  x + y                  ≈⟨ sym $ cong₂ _+_ (fromℕ-toℕ x) (fromℕ-toℕ y) ⟩
+  fromℕ m + fromℕ n      ≈⟨ sym (fromℕ-homo-+ m n) ⟩
+  fromℕ (m ℕ.+ n)        ≤⟨ fromℕ-mono-≤ (ℕₚ.+-mono-≤ m≤m' n≤n') ⟩
+  fromℕ (m' ℕ.+ n')      ≈⟨ fromℕ-homo-+ m' n' ⟩
+  fromℕ m' + fromℕ n'    ≈⟨ cong₂ _+_ (fromℕ-toℕ x') (fromℕ-toℕ y') ⟩
+  x' + y'                ∎
+  where
+  open IneqReasoning
+  m    = toℕ x;             m'   = toℕ x'
+  n    = toℕ y;             n'   = toℕ y'
+  m≤m' = toℕ-mono-≤ x≤x';   n≤n' = toℕ-mono-≤ y≤y'
+
++-monoˡ-≤ :  ∀ x → (_+ x) Preserves _≤_ ⟶ _≤_
++-monoˡ-≤ x {y} {z} y≤z =  +-mono-≤ {y} {z} {x} {x} y≤z (≤-refl {x})
+
++-monoʳ-≤ :  ∀ x → (x +_) Preserves _≤_ ⟶ _≤_
++-monoʳ-≤ x {y} {z} y≤z =  +-mono-≤ {x} {x} {y} {z} (≤-refl {x}) y≤z
+
++-mono-<-≤ :  _+_ Preserves₂ _<_ ⟶ _≤_ ⟶ _<_
++-mono-<-≤ {x} {x'} {y} {y'} x<x' y≤y' =  begin-strict
+    x + y                     ≈⟨ sym $ cong₂ _+_ (fromℕ-toℕ x) (fromℕ-toℕ y) ⟩
+    fromℕ m + fromℕ n      ≈⟨ sym (fromℕ-homo-+ m n) ⟩
+    fromℕ (m ℕ.+ n)          <⟨ fromℕ-mono-< (ℕₚ.+-mono-<-≤ m<m' n≤n') ⟩
+    fromℕ (m' ℕ.+ n')        ≈⟨ fromℕ-homo-+ m' n' ⟩
+    fromℕ m' + fromℕ n'    ≈⟨ cong₂ _+_ (fromℕ-toℕ x') (fromℕ-toℕ y') ⟩
+    x' + y'                  ∎
+  where
+  open IneqReasoning
+  m    = toℕ x;             n    = toℕ y
+  m'   = toℕ x';            n'   = toℕ y'
+  m<m' = toℕ-mono-< x<x';   n≤n' = toℕ-mono-≤ y≤y'
+
++-mono-≤-< :  _+_ Preserves₂ _≤_ ⟶ _<_ ⟶ _<_
++-mono-≤-< {x} {x'} {y} {y'} x≤x' y<y' =  subst₂ _<_ (+-comm y x) (+-comm y' x') y+x<y'+x'
+  where
+  y+x<y'+x' =  +-mono-<-≤ {y} {y'} {x} {x'} y<y' x≤x'
+
++-monoˡ-< :  ∀ x → (_+ x) Preserves _<_ ⟶ _<_
++-monoˡ-< x {y} {z} y<z =  +-mono-<-≤ {y} {z} {x} {x} y<z (≤-refl {x})
+
++-monoʳ-< :  ∀ x → (x +_) Preserves _<_ ⟶ _<_
++-monoʳ-< x {y} {z} y<z =  +-mono-≤-< {x} {x} {y} {z} (≤-refl {x}) y<z
+
+suc-mono-≤ :  suc Preserves _≤_ ⟶ _≤_
+suc-mono-≤ {x} {y} x≤y =  begin
+  suc x     ≈⟨ suc≗1+ x ⟩
+  1B + x    ≤⟨ +-monoʳ-≤ 1B {x} {y} x≤y ⟩
+  1B + y    ≈⟨ sym (suc≗1+ y) ⟩
+  suc y     ∎
+  where
+  open IneqReasoning
```

Comment:
Please put the `open` statements on the same line as the `where`, here and below.